pubsub.go

```diff
@@ -98,6 +100,24 @@ type PubSub struct {
 	seenMessagesMx sync.Mutex
 	seenMessages   *timecache.TimeCache
 
+	// discovery assists in discovering and advertising peers for a topic
+	discovery discovery.Discovery
+
+	// advertising tracks which topics are being advertised
+	advertising map[string]context.CancelFunc
+
+	// bootstrapped tracks which topics have been bootstrapped
+	bootstrapped sync.Map
+
+	// rediscover handles continuing peer rediscovery
+	rediscover chan *Rediscover
```

Comment:
`rediscover` is a silly name, let's call this just `discover`.

Reply:
ok